data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetState.java

```diff
@@ -0,0 +1,122 @@
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.EXECUTION_OPTIMISTIC;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.FINALIZED;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.HEADER_CONSENSUS_VERSION;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_BEACON;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.BOOLEAN_TYPE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Optional;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.schema.Version;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+import tech.pegasys.teku.spec.datastructures.metadata.StateAndMetaData;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionCache;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitions;
+
+public class GetState extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v1/beacon/states/{state_id}";
+  private final ChainDataProvider chainDataProvider;
+
+  public GetState(
+      final DataProvider dataProvider, final SchemaDefinitionCache schemaDefinitionCache) {
+    this(dataProvider.getChainDataProvider(), schemaDefinitionCache);
+  }
+
+  public GetState(
+      final ChainDataProvider chainDataProvider,
+      final SchemaDefinitionCache schemaDefinitionCache) {
+    this(chainDataProvider, schemaDefinitionCache, ROUTE, "getStateV1", TAG_BEACON, false);
+  }
+
+  public GetState(
+      final ChainDataProvider chainDataProvider,
+      final SchemaDefinitionCache schemaDefinitionCache,
+      final String route,
+      final String operationId,
+      final String tag,
+      final boolean deprecated) {
+    super(
+        EndpointMetadata.get(route)
+            .operationId(operationId)
+            .summary("Get full BeaconState object")
+            .description(
+                "Returns full BeaconState object for given state_id.\n\n"
+                    + "Use Accept header to select `application/octet-stream` if SSZ response type is required.")
+            .tags(tag)
+            .deprecated(deprecated)
+            .pathParam(PARAMETER_STATE_ID)
+            .response(
+                SC_OK,
+                "Request successful",
+                getResponseType(schemaDefinitionCache, tag),
+                sszResponseType())
+            .withNotFoundResponse()
+            .build());
```

Comment:
this is probably cleaner if we have a static function for getEndpointMetadata and each class just implements theirs, and the constructor calls that (probably protected) function to get metadata...

then we're not passing flags like tag, operation (deprecated!) etc. it should wash up in a similar way but be a lot cleaner i would expect...

A lot of our endpoints i think had static functions, this would be a similar idea but doesnt need to be static and can be overridden....